simpleclient/src/main/java/io/prometheus/client/Counter.java

```diff
@@ -135,6 +137,24 @@ public void inc(double amt) {
       }
       value.add(amt);
     }
+    /**
+     * Increment the counter by given amount and update the exemplar for the series.
+     *
+     * @param amt the amount to increment the counter by
+     * @param exemplarLabels a set of key value pairs used to create the exemplar
+     *
+     * @throws IllegalArgumentException If amt is negative.
+     * @throws IllegalArgumentException if exemplarLabels is null.
+     * @throws IllegalArgumentException if exemplarLabels does not contain an even number of values.
+     * @throws IllegalArgumentException if exemplarLabels contains any invalid labels.
+     * @throws IllegalArgumentException if the combined length of the exemplarLabels is greater than 64.
+     */
```

Comment:
Initial golang documentation that I can probably borrow from a little more heavily:
https://github.com/prometheus/client_golang/blob/c32ffd121f40843d8ce747536467bb091678fd87/prometheus/counter.go#L44-L51

There were two things I wanted to point out here.

Do we want to copy the go behavior of ignoring null labels?
I'm inclined to say that null labels are invalid and that the user should use `inc` instead.

Exceptions for invalid labels scare me a bit as well. Having instrumentation code throw an exception  because your dynamically generated label ended up a little too long sounds like a not fun user experience, but I don't have a better alternative other than logging an error or incrementing an internal error counter.